### DISCUSSION TITLE
Active monitoring and agents?

### DISCUSSION THREAD

**Discussion by Skyline643** [Ylianst/MeshCentral#6542] [Ideas]
First of all, I would like to mention how great MeshCentral is! I use it for 3 years now, did some tweaks for the new icons, add MFA etc.

I know MeshCentral is more an alternative to Teamviewer or Anydesk (and way better). There's a 7 day monitoring but no active monitoring). Is there a way to get a message (by mail) when a device is down? When you add users, you need to add an e-mail address as well, but it's just adding. The mail address is not used for anything I think (maybe in the background or for MFA)?

Is there a possibility to expand the offline/online calender from 7 days to 14 days or 1 month?

We use it to monitor devices over the internet. The agent runs on Ubuntu/Windows devices and connects to a DNS address. Maybe an idea (but als complicated I think) to add sub-devices on th other side. So the agent is looking or monitoring other devices like a switch or something (just a ping to an IP would be great). Like a treaa you can expand and see which devices are under the "main" device.

I'm just thinking about some things that would be very useful in our case.

**Comment by si458** [comment]
Each user needs an email address adding as well as smtp setup on meshcentral 

Then each user display selects which devices they want notification and they get the emails :)

**Comment by Skyline643** [comment]
Ok thanks. I found how to setup mailing in the config.json. Users have already an e-mail address.

But I don't know exactly what lines I need to add in the config file, because we use MFA on Office 365. Can I add a token or how does the config look (especially the part for MFA)?

**Reply by si458** [reply]
erm to use office 365 with MFA, i believe u need to check in your portal u have SMTP enabled for your org
then u need to create an APP PASSWORD for the user you want to use to send the emails

**Reply by Skyline643** [reply]
Ok, I will try this out

**Comment by Skyline643** [comment]
I finally found some time to configure the mail notifications but I'm a bit stuck.
I'm always getting this:
![image](https://github.com/user-attachments/assets/434358a4-b3e6-4469-b372-a7b88bcd33a7)

However my config file looks like this:
![image](https://github.com/user-attachments/assets/4251e208-4c37-402e-a3d3-b3a689f85ad5)

Restarting the service doesn't matter to get this working.

What is missing in my configuration?

**Reply by si458** [reply]
You have an underscore set infront of smtp
Anything with an underscore in front of is ignored
So simply remove the underscore then restart

**Comment by Skyline643** [comment]
Yes indeed, you're right.
Seems to work, except I need to validate with a verification e-mail. If I uncomment the verifyEmail tag, the service is down. Seems to be a wrong syntax I think?

**Reply by si458** [reply]
all verifyemail does is `When set to false, the email format and DNS MX record are not checked.`
so if true (default) it will just verify the domains in your email have valid MX records to receive emails

**Comment by Skyline643** [comment]
Ok, got everything working to send mails via our firewall.
Is there also a possibility to add more information in the mail itself? Or to adjust the message that's been sent?
I see that MeshCentral is sending a mail when a device is offline for 5 minutes, not earlier. That's ok, but I don't know if I can change this as well if we want to in the future?

**Reply by si458** [reply]
so to change the emails templates
1. create a folder `meshcentral-web` which sits next to `meshcentral-data`
2. go into `meshcentral-web` and create a folder `emails`
3. copy the folder contents of `node_modules/meshcentral/emails/` to `meshcentral-web/emails/`
4. edit the ENGLISH files in the new folder to what you would like (both html and txt)
5. dont forget to edit the files inside of `translations` too, if you want to use multiple languages
6. once finished, restart meshcentral and it will use your templates, even after new versions of meshcentral are released
 
as for the email notifications,
the way the notification system is designed is it pools notifications together,
so a device goes offline, the notification gets put into a pool with other notifications 
then every 5 mins it will send a single notification with like a summary of whats happened in those 5 mins

we could possible add an option in to let the server admins change the 5mins to say every X minutes they want
OR
maybe to send notifications for every single event straight away
but say a site goes offline, you wouldnt get 1 notificiation with a summary of X num of comps went off
you would get 50 notifications with each saying THIS COMP went off, which would be a STUPID idea personally

**Comment by Skyline643** [comment]
Ok thanks for the information. The summary is a good idea indeed. Much better than a 100 mails. The frequency of 5 minutes is good enough. We can live with that. I was just curious how it was programmed :-)
